test/IdentityManager.js

```diff
@@ -0,0 +1,395 @@
+const lightwallet = require('eth-signer')
+const evm_increaseTime = require('./evmIncreaseTime.js')
+const snapshots = require('./evmSnapshots.js')
+const IdentityManager = artifacts.require('IdentityManager')
+const Proxy = artifacts.require('Proxy')
+const TestRegistry = artifacts.require('TestRegistry')
+const Promise = require('bluebird')
+const compareCode = require('./compareCode')
+web3.eth = Promise.promisifyAll(web3.eth)
+
+const LOG_NUMBER_1 = 1234
+const LOG_NUMBER_2 = 2345
+
+const userTimeLock = 100;
+const adminTimeLock = 1000;
+const adminRate = 200;
+
+contract('IdentityManager', (accounts) => {
+  let proxy
+  let deployedProxy
+  let testReg
+  let identityManager
+  let user1
+  let user2
+  let user3
+  let user4
+  let user5
+  let nobody
+
+  let recoveryKey
+  let recoveryKey2
+
+  let snapshotId
+
+  before(done => {
+    // Truffle deploys contracts with accounts[0]
+    user1 = accounts[0]
+    nobody = accounts[1] // has no authority
+    user2 = accounts[2]
+    user3 = accounts[3]
+    user4 = accounts[4]
+    user5 = accounts[5]
+    recoveryKey = accounts[8]
+    recoveryKey2 = accounts[9]
+    IdentityManager.new(userTimeLock, adminTimeLock, adminRate).then((instance) => {
+      identityManager = instance
+      return Proxy.new({from: user1})
+    }).then((instance) => {
+      deployedProxy = instance
+      return TestRegistry.deployed()
+    }).then((instance) => {
+      testReg = instance
+    //   return snapshots.snapshot()
+    // }).then(id => {
+    //   snapshotId = id
+      done()
+    })
+  })
+
+  // afterEach(done => {
+  //   snapshots.revert(snapshotId)
+  //   done()
+  // })
+
+  it('Correctly creates Identity', (done) => {
+    let log
+    identityManager.CreateIdentity(user1, recoveryKey, {from: nobody}).then(tx => {
+      log = tx.logs[0]
+      assert.equal(log.event, 'IdentityCreated', 'wrong event')
+
+      assert.equal(log.args.owner,
+                   user1,
+                   'Owner key is set in event')
+      assert.equal(log.args.recoveryKey,
+                   recoveryKey,
+                   'Recovery key is set in event')
+      assert.equal(log.args.creator,
+                   nobody,
+                   'Creator is set in event')
+
+    // TODO find out why this all of a sudden doesn't work
```

Comment:
I think this probably has something to do with which version of solc truffle is using.